ash/src/extensions/amdx/shader_enqueue.rs

```diff
@@ -38,7 +38,9 @@ impl ShaderEnqueue {
             allocation_callbacks.as_raw_ptr(),
             pipelines.as_mut_ptr(),
         )
-        .result_with_success(pipelines)
+        .result()?;
+        pipelines.set_len(create_infos.len());
```

Comment:
Should we have a `set_len_on_success<T>(self, v: Vec<T>, len: usize) -> Result<Vec<T>>` helper?

Reply:
Sounds like a good idea, maybe we can share some code with other vector initializers, I'll have a look at that. Even cooler would be something utilizing `Vec<MaybeUninit<T>>` but `with_capacity()` serves the same purpose for now.

Reply:
> Even cooler would be something utilizing Vec<MaybeUninit<T>>

I think that's specifically redundant to `Vec`'s notion of capacity, yeah. We could do `Box<[MaybeUninit<T>]>` and convert that to `Box<[T]>` but that seems needlessly esoteric.

Reply:
Fair point, there are even helper functions to get the uninitialized part of a `Vec` as `&mut [MaybeUninit<T>]` but that's not really of use to us.

Note that a similar call for pipelines in `device.rs` was returning this `set_len`'d `Vec` even when `create_graphics_pipelines()`/`create_compute_pipelines()` returns `Err`. This is because some of the pipelines could have been created, but I don't remember if the driver sets the rest to `VK_NULL_HANDLE` or if we should initialize it to that.